src/map/pc.cpp

```diff
@@ -7046,9 +7046,10 @@ char pc_randomwarp(map_session_data *sd, clr_type type, bool ignore_mapflag)
 	if (mapdata->getMapFlag(MF_NOTELEPORT) && !ignore_mapflag) //Teleport forbidden
 		return 3;
 
+	int edge = battle_config.map_edge_size;
 	do {
-		x = rnd()%(mapdata->xs-2)+1;
-		y = rnd()%(mapdata->ys-2)+1;
+		x = rnd()%(mapdata->xs - edge*2) + edge;
+		y = rnd()%(mapdata->ys - edge*2) + edge;
 	} while((map_getcell(sd->bl.m,x,y,CELL_CHKNOPASS) || (!battle_config.teleport_on_portal && npc_check_areanpc(1,sd->bl.m,x,y,1))) && (i++) < 1000);
```

Comment:
Should we decrease from 1000 to 50 here too?

Reply:
We could, but it doesn't make a big difference since the chance to even reach 50 attempts is low.

Reply:
I talked with Daegalagh about this before, I think long term we should unify all into using the same function(s). Even monsters using teleport should actually not land on portals, so it would be nice to have this check in the map_search_freecell function anyway. I just don't want to do this as part of this PR (if you change a lot of functionality and at the same time do a full refactoring, it's hard to see what functionality was changed).